src/packages/bem/README.md

```diff
@@ -0,0 +1,232 @@
+.ComponentName {}
+/* component's root node class name with boolean modifyer applied */
+.ComponentName--modName {}
+/* component's root node class name with string/number modifyer applied */
+.ComponentName--modName_modValue {}
+/* component's sub node (element) class name */
+.ComponentName__elem {}
+/* component's root node class name with boolean modifyer + value applied */
+.ComponentName__elem--modName {}
+/* component's root node class name with string/number modifyer + value applied */
+.ComponentName__elem--modName_modValue {}
+```
+
+Example of usage
+----------------
+
+### Statefull component
+
+Button.js
+```js
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+// 1. Don't forget to import `bem` decorator
+import bem from 'bem';
+// 2. Import classnames map (webpack's css loader should be configured in css-module mode)
+import styles from './Button.scss';
+
+// 3. Annotate component with `bem` decorator and pass styles to it
+@bem(styles)
+export default class Button extends Component {
+
+    static propTypes = {
+        active: PropTypes.bool,
+        type: PropTypes.oneOf([ 'normal', 'extraordinary' ]),
+    };
+    static defaultProps = {
+        active: false,
+        type: 'normal',
+    };
+
+    // 4. If you need to have class name (.Button--active) that depends on
+    //    `active` prop, just list this prop in propsToMods list.
+    static propsToMods = ['active'];
+
+    // 5. If you need to have class name (.Button--clicked) that depend on
+    //    state.clicked value, just list this prop in propsToMods list.
+    static stateToMods = ['clicked'];
+
+    state = { clicked: false };
+
+    handleClick = () => {
+        const { clicked } = this.state;
+        this.setState({ clicked: !clicked });
+    };
+
+    render() {
+        return (
+            {/*  6. Add { ...this.block() } construction to declare node as a block root */}
+            <button { ...this.block() } onClick={this.handleClick}>
+                {/*  7. Add { ...this.elem('label') } construction to declare node as a label element */}
+                <span { ...this.elem('label') }>
+                    {this.props.children}
+                </span>
+            </button>
+        );
+    }
+}
+```
+
+ButtonStateless.js
+```js
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Button.scss';
+
+// 3. Annotate component with `bem` decorator and pass styles to it
+import React from 'react';
+
+function ButtonStateless(props) {
+    return (
+      {/*  1. Add { ...props.block() } construction to declare node as a block */}
+      <button { ...props.block() }>
+        {/*  2. Add { ...props.elem('label') } construction to declare node as a label element */}
+        <span { ...props.elem('label') }>
+            {props.children}
+        </span>
+      </button>
+    );
+);
+
+ButtonStateless.displayName = 'Button';
```

Comment:
Here it's not necessary, but it's an example how you can set block name (Button) that is different from function name (ButtonStateless). I'll add a comment to make it explicit.

`displayName` is necessary though if you declare component like this:
```js
const someComponent = () => <div/>;
someComponent.displayName = 'someComponent';

// Also, this kind of declaration gives no chance to find out block name:
export default () => <div/>;
```

Reply:
I never had to declare `displayName` before even when creating components with arrow function. Anyway, I think this is not relevant to BEM and just add distractions here. It is just general React usage

Reply:
Actually, it is relevant to BEM, because bem.js should be able to somehow find out a name of the component. Now it tries to take displayName. If displayName is not set, then it looks for function or class name.

But in case of anonymous function there is no way to find out a name.

Reply:
I see. Can you add this to the docs then? That would mean that we have to define it all the time manually if we use arrow functions for component declaration.

Reply:
Ok, I've checked and it indeed works without displayName. Updated docs correspondently.